src/Processors/Formats/Impl/ParquetBlockInputFormat.cpp

```diff
@@ -450,6 +474,28 @@ ParquetBlockInputFormat::~ParquetBlockInputFormat()
         pool->wait();
 }
 
+std::shared_ptr<parquet::FileMetaData> ParquetBlockInputFormat::getFileMetaData()
+{
+    if (!use_metadata_cache || !metadata_cache_key.length())
+    {
+        ProfileEvents::increment(ProfileEvents::ParquetMetaDataCacheMisses);
```

Comment:
Do we really want metadata cache miss metrics being incremented when `use_metadata_cache=false`?

Reply:
I coded the increment in the first pass, but removed it so that users need not see a new **alarming** "miss" metric if they are unaware/disabled the in-memory cache feature (and satisfied with the existing local disk caching feature). Let me know what you think.

Reply:
I am sorry, I did not quite follow your explanation. As far as I can tell (by reading the code), whenever `initializeIfNeeded()` is called (and it is called once per file, iirc), it'll try to fetch metadata either from cache or from input source.

If cache is disabled, it'll still increment the cache miss. It doesn't look right to me at a first glance, but if there has been a discussion and this was the chosen design, that's ok.

Am I missing something?

Reply:
Correct, I mixed up my revisions and reasoning. I have removed the increment if cache is disabled. Thanks!